File: eds-starter6-simple-mongodb/client/app/model/Department.js
Ext.define("Starter.model.Department",
{
  extend : "Ext.data.Model",
  requires : [ "Ext.data.proxy.Direct" ],
  fields : [ {
    name : "id",
    type : "string"
  }, {
    name : "name",
    type : "string"
  } ],
  proxy : {
    type : "direct",
    directFn : "departmentService.read",
    writer : {
      writeAllFields : true
    }
  }
});

File: eds-starter6-simple-jpa/src/main/java/ch/rasc/eds/starter/entity/Department.java
package ch.rasc.eds.starter.entity;

import javax.persistence.Entity;

import org.springframework.data.jpa.domain.AbstractPersistable;

import ch.rasc.extclassgenerator.Model;

@Model(value = "Starter.model.Department", readMethod = "departmentService.read")
@Entity
public class Department extends AbstractPersistable<Long> {

	private String name;

	public String getName() {
		return this.name;
	}

	public void setName(String name) {
		this.name = name;
	}

}
